backend/danswer/server/manage/users.py

```diff
@@ -213,6 +213,50 @@ def deactivate_user(
     db_session.commit()
 
 
+@router.delete("/manage/admin/delete-user")
+async def delete_user(
+    user_email: UserByEmail,
+    _: User | None = Depends(current_admin_user),
+    db_session: Session = Depends(get_session),
+) -> None:
+    user_to_delete = get_user_by_email(
+        email=user_email.user_email, db_session=db_session
+    )
+    if not user_to_delete:
+        raise HTTPException(status_code=404, detail="User not found")
+
+    if user_to_delete.is_active is True:
+        logger.warning(
+            "{} must be deactivated before deleting".format(user_to_delete.email)
+        )
+        raise HTTPException(
+            status_code=400, detail="User must be deactivated before deleting"
+        )
+
+    # Detach the user from the current session
+    db_session.expunge(user_to_delete)
+
+    try:
+        # Delete related OAuthAccounts first
+        for oauth_account in user_to_delete.oauth_accounts:
+            db_session.delete(oauth_account)
+
+        db_session.delete(user_to_delete)
+        db_session.commit()
+
+        user_emails = get_invited_users()
```

Comment:
Probably a race condition if two calls to this api are made at the same time?  Let's add a note to the code here and a fix it ticket.